core/src/main/java/net/consensys/eventeum/chain/contract/ConfirmationCheckInitialiser.java

```diff
@@ -28,22 +31,33 @@
 
     private ChainServicesContainer chainServicesContainer;
     private BlockchainEventBroadcaster eventBroadcaster;
-    private EventConfirmationConfig eventConfirmationConfig;
     private AsyncTaskService asyncTaskService;
+    private NodeSettings nodeSettings;
 
     @Override
     public void onEvent(ContractEventDetails eventDetails) {
         if (eventDetails.getStatus() == ContractEventStatus.UNCONFIRMED) {
             log.info("Registering an EventConfirmationBlockListener for event: {}", eventDetails.getId());
 
             final BlockchainService blockchainService = getBlockchainService(eventDetails);
-            blockchainService.addBlockListener(createEventConfirmationBlockListener(eventDetails));
+            final Node node = nodeSettings.getNode(blockchainService.getNodeName());
+            BigInteger currentBlock = blockchainService.getCurrentBlockNumber();
+            BigInteger waitBlocks = node.getBlocksToWaitForConfirmation();
+
+            if (currentBlock.compareTo(eventDetails.getBlockNumber().add(waitBlocks)) >= 0) {
```

Comment:
We can't assume that the event is confirmed just because the block number has been reached.  We also need to check that the tx is still included within the mined block.